libbeat/scripts/Makefile

```diff
@@ -313,13 +320,72 @@ install-home:
 	install -d -m 755 ${HOME_PREFIX}/scripts/
 	install -m 755 ${ES_BEATS}/libbeat/scripts/migrate_beat_config_1_x_to_5_0.py ${HOME_PREFIX}/scripts/
 
-# Creates packer files in a beat
-.PHONY: create-packer
-create-packer:
-	python ${ES_BEATS}/libbeat/scripts/create_packer.py --es_beats=${ES_BEATS} --beat_path=${BEAT_DIR} --beat=${BEATNAME}
+
+# Prepares for packaging. Builds binaries and creates homedir data
+.PHONY: prepare-package
+prepare-package: build/upload
+	# cross compile on ubuntu
+	docker run --rm \
+		-v $(abspath ${ES_BEATS}/dev-tools/packer/xgo-scripts):/scripts \
+		-v $(abspath ../):/source \
+		-v $(BUILD_DIR):/build \
+		-e PUREGO="yes" \
+		-e PACK=${BEATNAME} \
+		-e BEFORE_BUILD=before_build.sh \
+		-e SOURCE=/source \
+		-e TARGETS=${CROSSCOMPILE_TARGETS} \
+		-e BUILDID=${BUILDID} \
+		tudorg/beats-builder \
+		${BEAT_DIR}
+
+# Prepares for packaging. Builds binaries with cgo
+.PHONY: prepare-package-cgo
+prepare-package-cgo: build/upload
+
+	# TODO: rename crosscompile_targets are not accurate related to cgo -> it is only for linux builds on debian6
+
+	# cross compile on ubuntu
+	docker run --rm \
+		-v $(abspath ${ES_BEATS}/dev-tools/packer/xgo-scripts):/scripts \
+		-v $(abspath ../):/source \
+		-v $(BUILD_DIR):/build \
+		-e PACK=${BEATNAME} \
+		-e BEFORE_BUILD=before_build.sh \
+		-e SOURCE=/source \
+		-e TARGETS=${CROSSCOMPILE_TARGETS} \
+		-e BUILDID=${BUILDID} \
+		tudorg/beats-builder \
+		${BEAT_DIR}
+
+	# linux builds on debian 6 for compatibility
+	docker run --rm \
+		-v ${BUILD_DIR}:/build \
+		-v $(abspath ${ES_BEATS}/dev-tools/packer/xgo-scripts):/scripts \
+		-v $(abspath ..):/source \
+		-e PACK=${BEATNAME} \
+		-e BEFORE_BUILD=before_build.sh \
+		-e SOURCE=/source \
+		-e TARGETS=${CROSSCOMPILE_TARGETS_CGO} \
+		-e BUILDID=${BUILDID} \
+		tudorg/beats-builder-deb6 \
+		${BEAT_DIR}
+
+include $(ES_BEATS)/dev-tools/packer/scripts/Makefile
+
 
 # Create binary packages for the beat
-package: create-packer
-	cd dev-tools/packer; \
-	export ES_BEATS=${ES_BEATS} ; \
-	make deps images all
+package:
+	if [ $(CGO) = true ]; then \
+		 $(MAKE) prepare-package-cgo; \
+	else \
+		$(MAKE) prepare-package; \
+	fi
+
+	# Generates the package.yml file with all information needed to create packages
+	echo "beat_name: ${BEATNAME}" > ${BUILD_DIR}/package.yml
+	echo "beat_url: https://${BEAT_DIR}" >> ${BUILD_DIR}/package.yml
+	echo "beat_repo: ${BEAT_DIR}" >> ${BUILD_DIR}/package.yml
+	echo "beat_description: ${BEAT_DESCRIPTION}" >> ${BUILD_DIR}/package.yml
```

Comment:
I really like this idea as it reduces the complexity in the Makefile and keeps the packaging out of it, but at the same time allows us to do things that were not possible before.